### DISCUSSION TITLE
V6 - Argument of type '() => typeof Role' is not assignable

### DISCUSSION THREAD

**Discussion by imcm7** [mikro-orm/mikro-orm#4721] [General]
```
export class Author extends BaseEntity {
  @ManyToMany(() => Book, undefined, { pivotTable: 'book_to_tag_unordered', orderBy: { name: QueryOrder.ASC } })
}
```

Was
```
export class Author extends BaseEntity<Book, 'id'> {
```

>Argument of type '() => typeof Author' is not assignable to parameter of type 'string | ManyToManyOptions<Author, unknown> | (() => EntityName<Author>) | undefined'.
  Type '() => typeof Author' is not assignable to type '() => EntityName<Author>'.ts(2345)

**Comment by B4nan** [comment]
And what is `Book`? Maybe you used a type import for it?

**Reply by imcm7** [reply]
I understand, the entity from many to many was with `extends BaseEntity<Book, 'id'>` but need `extends BaseEntity`. I not updated all entities yet.

**Reply by B4nan** [reply]
This is not about base entities. What do you get if you `console.log(Book)` in that file where it fails? 

> export class Author extends BaseEntity<Book, 'id'> {

`Author` extends `BaseEntity<Book>`?

**Reply by B4nan** [reply]
Oh or you mean you upgraded to v6 and didn't change that? And removing the type arg helps?

**Reply by imcm7** [reply]
Yes? when I leave only `extends BaseEntity`, warning disappear. 
Will I have problems with that when will do queries like find, findAndCount and other?
